app/main.py

```diff
@@ -1 +1,45 @@
-#  write your code here
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 3, is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
```

Comment:
Why is this method static?

Reply:
The `catch_mouse()` method in the `Cat` class is marked as static with `@staticmethod` decorator because it doesn't need to access or modify any instance variables or methods. 

A static method is bound to the class and not to the instance of the class, meaning that it can be called on the class itself, without the need for an instance of that class. In this case, `catch_mouse()` method does not rely on any instance-specific information (such as the cat's name, hunger level, etc.), and so it can be defined as a static method.

Reply:
Logically, this method must be called specifically on the instance of the class. If we call this method on class it will look illogically.